jnius/reflect.py

```diff
@@ -195,6 +196,20 @@ def log_method(method, name, signature):
         Modifier.isStrict(mods)
     )
 
+def identifyHierarchy(cls, level, classList, concrete=True):
+    classList.append((cls,level))
+    supercls = cls.getSuperclass()
+    if supercls is not None:
+        identifyHierarchy(supercls, level+1, classList, concrete)
+    intfcs = cls.getInterfaces()
+    if intfcs is not None:
+        for intf in intfcs:
+            identifyHierarchy(intf, level+1, classList, concrete)
+    if not concrete and cls.isInterface() and len(intfcs) ==0:
+        classList.append((find_javaclass('java.lang.Object'),level+1))
+    #return classList
```

Comment:
thanks for the timely feedback. I can make the changes. 

The yield looks more succinct - I dont know if it will work in Python 2 or not, so to be clear, do you wish this change?

Reply:
i didn't test it, but i'm confident python2 shouldn't make a diff here, if we dropped python2 we could make it even more succint by replacing the inner loops with `yield from identify_hierarchy(_)`